### PR TITLE
User profile updates

### PR DESCRIPTION
**Pull request by pazoozoo42**
- option to show and edit profile bio
 - move theme toggle to the user profile page

Own profile:
image

image

Any other user profile:
image

image

### CODE REVIEW

File: frontend/src/Types/UserInfo.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ export type UserBaseInfo = {
     id: number;
     username: string;
     gender: UserGender;
+    bio_source: string;
```
Comment (Aivean) [inline frontend/src/Types/UserInfo.ts:11]: Perhaps `UserProfileInfo` is a better place for these fields?

File: frontend/src/Components/CreateCommentComponent.tsx [hunk 2]
```diff
@@ -205,16 +206,22 @@ export default function CreateCommentComponent(props: CreateCommentProps) {
...
+                        setPreviewing(response);
+                    }
+                }
             })
             .catch(error => {
                 console.log('onAnswer ERR', error);
             })
             .finally(() => {
-                setPreviewing(null);
+                //setPreviewing(null);
```
Comment (Aivean) [inline frontend/src/Components/CreateCommentComponent.tsx:224]: Why is this commented out?

Comment (pazoozoo42) [reply frontend/src/Components/CreateCommentComponent.tsx]: debug leftovers, thanks for catching this

**Review by Aivean — CHANGES_REQUESTED** [review]
Overall, looks good. 

One thing that I'm not certain of, is how the preview/editing is implemented in user profile UX-wise.

image



I have several concerns about current implementation:
* Your profile looks different for you compared to how others see it. 
    this is a problem if we consider profile to be a "vanity space"
* Editing of the profile info happens pretty rarely compared to visiting the profile
    currently lots of unnecessary or outright not working UI elements (are shown: editor, editor buttons, "send" button) when own profile is visited
* The UX for profile editing is different from the editing of posts/comments (when the "final" version is shown after "submit" button is pressed). Is there a reason for that?

I'd suggest to make the profile editing UX same as it is for comments/posts. Display the final html in the profile and a small "edit" button, which, when pressed, displays the editor. This should also simplify the code for the `CreateCommentComponent`.

File: backend/src/api/types/requests/UserProfile.ts [hunk 3]
```diff
@@ -45,4 +45,12 @@ export type UserRestrictionsResponse = {
...
-};
\ No newline at end of file
+};
+
+export type UserSaveBioRequest = {
+    bio: string;
+};
+
+export type UserSaveBioResponse = {
+    bio: string | boolean;
```
Comment (Aivean) [inline backend/src/api/types/requests/UserProfile.ts:55]: Please don't do this. Error code is already available on the http level.

**Pull request closed without merging**